### ISSUE TITLE
Quarkus CLI option for working with custom maven settings

### ISSUE THREAD

**Issue by fedinskiy** [quarkusio/quarkus#23506]
### Description

When working with Quarkus CLI, it may be desirable to have a separate maven settings file, but undesirable to have it as a default "/.m2/settings.xml". For that case, it would be convenient to have an option, which will be work in a manner similar to `mvn -s settings.xml`.  

### Implementation ideas

Several:
1. New option (`--maven-settings`) which accepts a path to a file, which is passed  to `mvn` command as `-s` option
2. New option (`--maven-options`) which accepts a line with some maven options, which passed directly to `mvn`(e.g. `quarkus create app newapp --maven-options "-s settings.xml -U -B"`)
3. Add options similar to 1. or 2. to `~/.quarkus/config.yaml` file

**Comment by rsvoboda** [comment]
Adjustments of `.mvn/maven.config` work for generated project. But how to handle `quarkus create app myapp` case?

**Comment by rsvoboda** [comment]
Well, are maven artifacts needed for `quarkus create app myapp`? Maybe for codestarts, right?

**Comment by ebullient** [comment]
the usual maven environment variables will also work. MAVEN_OPTS / MAVEN_ARGS

**Comment by rsvoboda** [comment]
@fedinskiy please check if provided info makes your use-case possible now.

About MAVEN_ARGS, this is not working for me (using registry.quarkus.redhat.com, redhat maven repo enabled in my custom settings, but `~/.m2/settings.xml` still gets picked 
```
MAVEN_ARGS="-s settings.xml -U -B" quarkus create app myapp

[ERROR] ❗  Unable to create project: Failed to resolve Quarkus extension catalog com.redhat.quarkus.platform:quarkus-bom-quarkus-platform-descriptor:json:2.2.3.SP2-redhat-00001:2.2.3.SP2-redhat-00001 from central (https://repo.maven.apache.org/maven2)
```

https://maven.apache.org/configure.html#maven_args-environment-variable says it's Maven 4 thing

**Comment by rsvoboda** [comment]
@aloubyansky I think this is resolved in https://github.com/quarkusio/quarkus/pull/24485

`propagateProperty("maven.settings", mavenSettings, newArgs);` should handle that

**Comment by aloubyansky** [comment]
It might be with `-Dmaven.settings`. I'll need to check all the commands. However, I think `--maven-settings` would be a better option.

**Comment by aloubyansky** [comment]
I actually thought of that too.

**Comment by ebullient** [comment]
We were (at some point) going to allow more interesting quarkus configuration.
The other option (that I would prefer to another argument) is picking up maven settings from the quarkus configuration (so you only have to specify one config, with the option we already have)

**Comment by aloubyansky** [comment]
@ebullient do you still agree `--maven-settings` would be useful though?

**Comment by ebullient** [comment]
no. feels like command line argument creep to me given we already have a config file (that we should use for these things).

**Comment by maxandersen** [comment]
Fwiw jbang honors maven.setting for this exact usecase (be able to control/share build settings)

I really don't grok why we would want to have settings in config we can't override on command line. 

Having to change config file for special case setting does not make sense to me ?

**Comment by ebullient** [comment]
I guess outside of us testing, I don't see this as a one-off occurrence. If I'm working w/ a company firewall, that repository choice will be persistent.. and if I use gradle, creating a maven settings file for that seems odd?
